### PR TITLE
bump: Upgrade do cluster to 1.24

### PR DESCRIPTION
**Pull request by tfrgoncalves**
- Bump k8s cluster version from 1.23 to 1.24
- Add terraform plan step to ci
- Add validation to _k8s_cluster_id_ terraform variable, sanity check for tf plan rework
- Bump terraform k8s provider from 1.13.x to 2.20.x
- Bump terraform digital ocean provider from 2.23.x to 2.28.x
- Remove not used k8s provider in db deployment

### CODE REVIEW

File: .do/db-cluster/Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ TF ?= $(TF_EXTRA_OPTS) terraform
 WORKSPACE_PREFIX ?= postgres
 VARS_FILE ?= deploy.tfvars
 K8S_CLUSTER_REGION ?= fra1
-K8S_CLUSTER_ID ?= $(shell cat ../k8s-cluster/terraform_outputs.json | jq -r '.cluster_id.value')
+K8S_CLUSTER_ID ?= $(shell cat ../k8s-cluster/terraform_outputs.json 2>/dev/null | jq -r '.cluster_id.value' || True && echo 'REPLACE_ME')
```
Comment (tfrgoncalves) [inline .do/db-cluster/Makefile:9]: Can you think of any improvement to do here?

Comment (h314to) [reply .do/db-cluster/Makefile:9]: how about something like this:
```
doctl kubernetes cluster get codacy-doks-cluster -o json | jq -r '.[].id'
```

File: .do/db-cluster/variables.tf [hunk 2]
```diff
@@ -16,6 +16,10 @@ variable "k8s_cluster_id" {
   description = "The id of the k8s cluster where db connections will come from."
   type = string
   default = "REPLACE_ME"
```
Comment (h314to) [inline .do/db-cluster/variables.tf:18]: You could also refactor this to have as input the cluster name, and then use a [data source](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/data-sources/kubernetes_cluster) to get the cluster, from which you can extract its data, namely, its id

**Review by h314to — DISMISSED** [review]
Looking good, overall. Just left a few comments regarding how to get the cluster id, that may make this more robust.

File: .do/db-cluster/variables.tf (hunk 2 above)
Comment (tfrgoncalves) [reply .do/db-cluster/variables.tf:18]: Thought it would be preferable to keep using the same strategy.

File: .do/db-cluster/Makefile (hunk 1 above)
Comment (tfrgoncalves) [reply .do/db-cluster/Makefile:9]: Will employ the other approach, thank you.

File: .do/db-cluster/variables.tf (hunk 2 above)
Comment (tfrgoncalves) [reply .do/db-cluster/variables.tf:18]: Implemented.

**Pull request merged**